File: c++/2422.cpp
#include <iostream>
#include <vector>

using namespace std;

int n, m;
bool badCombi[201][201];
int answer = 0;
vector<int> v;

void dfs(int idx, int cnt) {
  if (cnt == 3) {
    answer++;
    return;
  }

  for (int i = idx; i <= n; i++) {
    bool flag = false;
    for (int j = 0; j < v.size(); j++) {
      if (badCombi[i][v[j]]) {
        flag = true;
        break;
      }
    }
    if (flag) continue;
    v.push_back(i);
    dfs(i + 1, cnt + 1);
    v.pop_back();
  }
}

int main() {
  freopen("example.txt", "r", stdin);

  cin >> n >> m;

  for (int i = 0; i < m; i++) {
    int a, b;
    cin >> a >> b;
    badCombi[a][b] = true;
    badCombi[b][a] = true;
  }

  dfs(1, 0);
  cout << answer;

  return 0;
}

File: js/2422.js
const fs = require('fs');

const inputs = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, m] = inputs[0].split(' ').map(Number);

const arr = [];
const badCombi = Array(n + 1)
  .fill()
  .map(() => Array(n + 1).fill(false));

let answer = 0;
for (let i = 1; i <= m; i++) {
  const [a, b] = inputs[i].split(' ').map(Number);
  badCombi[a][b] = true;
  badCombi[b][a] = true;
}

dfs(1, 0);
console.log(answer);

function dfs(idx, cnt) {
  if (cnt === 3) {
    answer++;
    return;
  }
  for (let i = idx; i <= n; i++) {
    if (!arr.some((el) => badCombi[i][el])) {
      arr.push(i);
      dfs(i + 1, cnt + 1);
      arr.pop();
    }
  }
}
